### ISSUE TITLE
Local Recall fails for different numbers of references

### ISSUE THREAD

**Issue by tuetschek** [GEM-benchmark/GEM-metrics#28]
It looks like Local Recall assumes that each example has the same number of references as the first one. That is not guaranteed.  @evanmiltenburg could you have a look at that?

Running:
```
./run_metrics.py --metric-list=local_recall -r test_data/single_dataset.refs.json test_data/single_dataset.outs.json
```

Triggers this error:
```
Traceback (most recent call last):
  File "./run_metrics.py", line 102, in <module>
    main(config)
  File "./run_metrics.py", line 69, in main
    values = gem_metrics.compute(outs, refs, srcs, metric_dict)
  File "/lnet/spec/work/people/odusek/gem-metrics/gem_metrics/__init__.py", line 100, in compute
    values.update(metric.compute(outs, refs))
  File "/lnet/spec/work/people/odusek/gem-metrics/gem_metrics/local_recall.py", line 27, in compute
    references.list_tokenized_lower_nopunct)
  File "/lnet/spec/work/people/odusek/gem-metrics/gem_metrics/local_recall.py", line 99, in local_recall_scores
    pair = (results[f'size-overlap-{n}'], results[f'size-refs-{n}'])
KeyError: 'size-overlap-4'
```

NB: You need to pull to trigger the error – I swapped the examples in the test file. I got the same error on DART, but the test file is faster.

**Comment by evanmiltenburg** [comment]
Fixed in 78bafa4. Simple change, but perhaps it would be good if you double-check that this indeed solves the issue (and the code still works as intended).

**Comment by tuetschek** [comment]
The code looks good and seems to work fine – thanks @evanmiltenburg !
